src/utils/logging.ts

```diff
@@ -0,0 +1,48 @@
+export interface Logging {
+  debug(name: string, msg: string): void;
+  warn(name: string, msg: string): void;
+  error(name: string, msg: string): void;
+  info(name: string, msg: string): void;
+}
+
+export enum LogLevel {
+  Debug,
+  Info,
+  Warn,
+  Error,
+}
+
+export class ConsoleLogger implements Logging {
+  constructor(public currentLogLevel: LogLevel, private throwOnError: boolean) {
+    this.currentLogLevel = currentLogLevel;
+    this.throwOnError = throwOnError;
+  }
+
+  private aboveLogLevel(loglevel: LogLevel): boolean {
+    return loglevel >= this.currentLogLevel ? true : false;
+  }
+
+  private formatMessage(name: string, msg: string) {
+    const currenttime = new Date().toLocaleString();
+    return `${currenttime} [${name}] ${msg}`;
```

Comment:
The time is a nice addition, but I think it'll be better without a date, and slightly more useful with milliseconds. So now it is:

```
29/05/2021, 17:30:52 [RP2040] Write to undefined address: 18000008
```

and I think something similar this will be more useful:

```
17:30:52.322 [RP2040] Write to undefined address: 18000008
```

Reply:
Done